### PR TITLE
111 cke rte editor

### PR DESCRIPTION
**Pull request by hbaptiste**
new cke adapter added as a plugin.
content, contribution and page applications have been modified in order to get the cke adapter to work.
These change will be committed later.

### CODE REVIEW

File: src/tb/apps/content/main.js [hunk 1]
```diff
@@ -78,7 +78,11 @@ define("app.content", ["tb.core", 'content.pluginmanager'], function (Core, Plug
         },
 
         enablePlugins: function () {
-            PluginManager.getInstance().enablePlugins();
+            try{
+                PluginManager.getInstance().enablePlugins();
+            }catch (e) {
+                console.log("... erreur ...");
```
Comment (ndufreche) [inline src/tb/apps/content/main.js]: console spotted.

File: src/tb/apps/content/plugins/rte/adapter/cke/main.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            }
+            var editorHtml = jQuery("#cke_" + e.editor.name);
+            if (jQuery(this.editorContainer).find(editorHtml).length) {
+                return;
+            }
+            jQuery(this.editorContainer).css({
+                padding: "5px",
+                width: "552px",
+                height: "90px"
+            });
```
Comment (ndufreche) [inline src/tb/apps/content/plugins/rte/adapter/cke/main.js]: It's necessary or you can use a class ?

File: src/tb/config.js [hunk 3]
```diff
@@ -72,14 +72,31 @@ define([], function () {
...
+                rte: {
+                    accept: ['*'],
+                    config: {
+                        adapter: "cke",
+                        aloha : {
+                            libPath: ''
+                        },
+                        cke: {
+                          libName: 'ckeeditor',
+                          libPath: 'src/tb/apps/content'//if provided where to find the library
```
Comment (ndufreche) [inline src/tb/config.js]: We really need the possibility for multi rte instance. One RTE by config, is enough I think.

File: src/tb/component/cke/skins/backbee/readme.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- **dev/**: contains SVG source of the skin icons.
+
+License
+-------
+
+Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+
+Licensed under the terms of any of the following licenses at your choice: [GPL](http://www.gnu.org/licenses/gpl.html), [LGPL](http://www.gnu.org/licenses/lgpl.html) and [MPL](http://www.mozilla.org/MPL/MPL-1.1.html).
+
+See LICENSE.md for more information.
```
Comment (ndufreche) [inline src/tb/component/cke/skins/backbee/readme.md]: You can remove this file.

File: src/tb/component/rtemanager/main.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        AbstactRteAdapter = new JS.Class({
+
+            initialize: function (config) {
+                this.config = config;
+                this.editor = null;
+                this.isEnabled = false;
+            },
+
+            onInit: function () {
+                throw "AbstactRteAdapterException:onInit - Not implemented yet";
```
Comment (ndufreche) [inline src/tb/component/rtemanager/main.js]: Use Core.Exception to centralise all the excepetions

File: src/tb/component/rtemanager/main.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            },
+
+            applyToContent: function () {
+                throw "AbstactRteAdapterException:applyToContent - Not implemented yet!";
```
Comment (ndufreche) [inline src/tb/component/rtemanager/main.js]: Use Core.Exception to centralise all the excepetions

File: src/require.config.js [hunk 7]
```diff
@@ -17,7 +17,7 @@
  * along with BackBuilder5. If not, see <http://www.gnu.org/licenses/>.
  */
 require.config({
-    baseUrl: 'resources/toolbar/',
+    baseUrl: '/resources/toolbar/',
```
Comment (ndufreche) [inline src/require.config.js]: Keep the relative url plz

**Comment by ndufreche** [comment]
Nice job, squash your commits and all will be good.
:+1:

**Comment by mickaelandrieu** [comment]
Great addition!

@hbaptiste: again, if you rebase master branch often, you won't suffer so much ;)

**Pull request merged**